### DISCUSSION TITLE
field reverts to initial value while submitting update

### DISCUSSION THREAD

**Discussion by ramseylove** [react-hook-form/react-hook-form#10321] [Q&A]
Do you know why this might be happening? The updated value reappears when submit was successful. See below code.

https://user-images.githubusercontent.com/3958672/233732903-aad1c97e-2d51-4279-bd72-1fcf451b950f.mov

Is there a setting that i should add to useForm?
```tsx
const endpointForm = useForm<Endpoint>({
    defaultValues: newEndpoint, // object with blank values for each field
    mode: 'all',
    values: data, // comes from api
  });
```

**Comment by leapful** [comment]
@ramseylove 

Based on your description, the root cause might be from your option `values: data` on `useForm` hook which will trigger re-rendering the whole form after submitting. Try to remove  this option to see if the issue is gone or not.

If `values` is the reason, just use `useEffect` and `setValues` to ensure that data api is only being used on form initialization.
